### DISCUSSION TITLE
Wino Calendar

### DISCUSSION THREAD

**Discussion by bkaankose** [bkaankose/Wino-Mail#85] [Ideas]
Hi everyone,

I want to start a new discussion about the next application that will be integrated into Wino Mail. I want to develop a calendar application and integrate it into Wino Mail, just like in Windows Mail & Calendars. It won't be a full complete replacement for Mail & Calendars without having calendar in place. This is also one of the most requested features from the users.

### App Architecture and Core Principles

- Core principles and priorities of the effort should not change; meaning that just like Wino Mail having similar design and functionality with Windows Mail, calendar will be the same Windows Calendar as much as possible. 
- All additional features that require extra efforts is delayed until app becomes stable/feature pair with Windows clone.
- Wino Mail and Calendar will share the same configurations and account setup. They will have their own isolated store and shared store for data sharing.
- Both applications will be separate apps, but apps may launch each other and communicate.
- Calendar must work without any account setup required; but appointments and meetings will be synchronized from Wino Mail if there are any.
- App is going to be a UWP + WinUI2, not WinAppSDK / WinUI 3.
- Startup performance, feature parity with Windows Calendar, stability is the top 3 priorities.

### Scope

Wino Mail had amazing feedback from both community members and other users so far. I wanted to shape it and prioritize development towards your feedback as much as I can. I think we came along a good way in just couple months already. 

Even though I use Windows Calendar for my personal computer, I am not aware of the full capability of this app. There will be/may be some functionalities that I'm not aware of. 

I want all of you to provide feedback as much as possible for Wino Calendar as well. Let's create the initial scope for this app and make it better for future.

You can reply under this discussion for all the initial things you wanted to see in Wino Calendar and additional stuff as well. Let's discuss and come up with the best solution for everybody. I will update this initial post and list them here as more suggestions provided by the users.

1. Calendar synchronization from Wino Mail.
2. Holiday calendars must be supported.
3. App must work without Wino Mail installed or any account setup in Wino Mail.
4. Printing calendar. Due to technical difficulties on this item it will be deprioritized for later stages after the first release.

**Comment by Benjin** [comment]
> App is going to be a UWP + WinUI2, not WinAppSDK / WinUI 3.

Can you tell me more about this decision?  I'm a bit out-of-date on the pros and cons of the different Windows app SDKs, but I'm curious why one might intentionally choose to target an older UX framework.

**Reply by bkaankose** [reply]
I personally find it more performant. For the record, I ported Wino Mail to WinAppSDK on a separate branch and it worked completely fine. Except:

- Worse startup performance
- Losing other UWP targets
- Couple UI glitches
- Lack of easy title bar customization options

It may feel an older framework, but I believe `nobody` can say that "WinAppSDK works much better than UWP". I don't see using an older framework as a disadvantage, actually it's an advantage for Wino's case. If ever WinAppSDK becomes performant as UWP, then migration can be done.

**Reply by lostmsu** [reply]
Have you tried alternatives like Avalonia?

I would love this app to be cross-platform because with the decisions Microsoft is making lately (like deprecating great Mail app) I feel eventual move to Linux is inevitable.

**Comment by Benjin** [comment]
Two simple features that I find extremely useful in Apple's calendar apps (especially in conjunction with each other), but aren't present in the Windows Calendar app: 1) duplicating an event, and 2) changing the calendar an event is on.

My scenario: I have a shared calendar with my partner, but also iCal feeds that I subscribe to (e.g. travel plans, sports schedule).  I very frequently want to copy an event (e.g. a "flight" event on my travel feed) to the feed I share with my partner so that she can see that info easily.

Note that I am not necessarily deleting the event from the source calendar (and may not have permissions to, in the event of a calendar feed or read-only source).

**Comment by Catscratch** [comment]
I think, for starters, it would be enought if WinoMail could handle appointments (ical) you get by mail. I'm using separate apps for mail and calendar which is working great. Only drawback of WinoMail (as replacement for WindowsMail) is that it can't handle ical events.

**Comment by albertyckwok** [comment]
Any news on this app. The native Win 10 is making it difficult to keep the classic calendar look and hoping we have something to similar to replace that.

**Comment by bkaankose** [comment]
With the full roll-off on Windows Mail & Calendars this got popular more than ever recently. I will see if I can prioritize this going forward.

**Comment by albertyckwok** [comment]
Thanks for trying. Please let me know if there is anything I can do to help. I tried to build and run Wino.Calendar.csproj but only see a blank screen. I haven't work on Windows coding for many years so it is likely that I am not doing many things correctly. I am using MS Visual Studio 2022 on Win 10.

**Reply by bkaankose** [reply]
It's a blank app :) I removed it from the UWP solution.

**Comment by AkazaRenn** [comment]
It would be great if there's widget support to replace Microsoft's terrible Outlook widget.

If it can collaborate with [Calendar Flyout](https://apps.microsoft.com/detail/9p2b3pljxh3v) that would be fantastic, too.

**Reply by Erofen** [reply]
Yes! The agenda view on hitting the date/time is SUPER important for me.

**Comment by sushinateur** [comment]
Hi there ! I wonder what the timeframe of the calendar would be. Looking for a replacement for the Windows-Calendar, maybe even as a stand-alone app. Any thoughts ? Thanks in advance !

**Comment by P40L0X** [comment]
Any news on the calendar integration?

**Reply by bkaankose** [reply]
I've already started the separation of common parts from Wino Mail last week. It's hard to give an estimate, but I can say at least work has been started.

**Comment by GreyManSoftware** [comment]
Will the calendar app be a standalone app like it was in Windows Mail and Calendar? I've become desperate for the calendar as Microsoft have started hard pushing everyone into Outlook and I can't face it! :D

**Comment by leo-tpsc** [comment]
I just want to say I'm very keen on this feature. I've wanted to use the app but can't due to the lack of calendar features. If I may ask, it would be great if we could also use it to schedule teams meetings as with the previous client. Thank you so much for working on this, so many of us can't stand the terrible new Outlook app and you might literally be the one saving Windows for us. Thank you so much!

**Comment by Y-PLONI** [comment]
I would greatly appreciate it if it could include a Hebrew calendar as well, with the option for recurring dates.

**Reply by Linho1219** [reply]
Not only the Hebrew calendar, but there are also many "local" calendars in the world, such as the Chinese lunar calendar, the Islamic calendar, and so on. Additionally, there are regions that use the Gregorian calendar but with different systems of reckoning years, such as Japan using the era name system and Taiwan using the Republic of China era. It would be great to support them all together.

**Comment by lostmsu** [comment]
I would greatly appreciate if the Calendar app is developed with some cross-platform framework like Avalonia instead of UWP, especially if it had Linux support (with Mail getting removed I am finally getting fed up with the ads and other spam MS lately shoves into my eyes, and considering jumping ship).

**Reply by AkazaRenn** [reply]
@lostmsu There are a lot of email clients on Linux why mess with one that's designed for Windows though?

**Reply by Benjin** [reply]
That... is a big request, lol.  Feels like a bit of a non-starter.

**Reply by lostmsu** [reply]
@Benjin what do you mean? UWP is not even being developed much by Microsoft anymore. It does not make sense to start a new project on it. Avalonia is very similar to UWP (and both of them to WPF). I am not asking for Linux support BTW, only that the app would be developed in a framework that is not Windows-only, so if somebody later would want to ensure it works on Linux or Mac they wouldn't have to rewrite the entire UI.

**Reply by AkazaRenn** [reply]
@lostmsu so you are asking the dev to rewrite the UI so that someone in the future we don't even know if exists who wants to port it to other platforms don't have to rewrite the UI am I taking it correctly?

**Reply by lostmsu** [reply]
@AkazaRenn what do you mean by rewrite? Calendar app is not developed yet.

**Reply by bkaankose** [reply]
Wino has its core libraries and view models abstracted. Both mail and calendar apps using the same core. This was initially done for code sharing, but it allows individual apps to be created on different platforms as well. However, I'm also against creating it with something other than UWP (plans for WinUI exists, but priority is finish the calendar with UWP first).

I'm no expert on Avalonia, which is a 3rd party UI framework. Uno suits better for Wino, but I don't find it suitable enough to create fast UIs.

**Reply by lostmsu** [reply]
The boss said their word.

**Reply by bkaankose** [reply]
What boss come on... :)

I started the initial work last month on the Calendar. For future plans it's much easier to develop with UWP first and then upgrade to WinUI for better support. @Tiktack already did some work on the AOT compatibility. End goal is to make both apps AOT WinUI.

**Comment by Esildor** [comment]
Have you considered open-sourcing the current state of the Calendar app? Would love to contribute some of my .NET skills!

**Reply by Tiktack** [reply]
Everything in a main branch. I am not sure how it actually works, but code is there :)

**Reply by Esildor** [reply]
Ah, ok, then I'd request any basic guides on how to contribute.

**Reply by bkaankose** [reply]
> Ah, ok, then I'd request any basic guides on how to contribute.

Lots of code has been changed since the initial contribution guidelines, but some parts are still relative. You can simply run the Wino.Calendar package to run server/calendar app. Right now there is no synchronization there.

**Comment by PeterHans-prog** [comment]
Calendar integration in Wino Mail is not important to me, as there are already two good alternatives to the Windows Mail & Calendar app and I use one of the calendars myself and am very happy with it.
For me, Wino Mail is important as a mail app and it has worked well, even if there are still areas for improvement and improvements.
A shared inbox would be great and the elimination of a few bugs. For example, I had a test email that could not be removed from the Yahoo Draft folder, even though it doesn't exist, it keeps appearing.
Keep it up, Wino Mail is great

**Reply by bkaankose** [reply]
Glad that you are enjoying Wino Mail so far. Yahoo uses IMAP and Wino Mail's IMAP synchronization will go under huge refactoring once the Calendar implementation is complete.

**Comment by PeterHans-prog** [comment]
***@***.***

Sent from Wino Mail<https://github.com/bkaankose/Wino-Mail/> for Windows


________________________________

**Comment by wetpaint2000** [comment]
Very much looking forward to the calendar integration.  Wino is superb, even more so with the calendar.

**Reply by Y-PLONI** [reply]
V

**Comment by wen-wen520** [comment]
Looking forward to the wino calendar, though I noticed that Wino.Calendar inside the repo, are there any new information can be shared about the calendar

**Comment by boccaccioe** [comment]
Curious if this is still in active development? Very excited about this release!

**Comment by bkaankose** [comment]
So many months and hours of development has been passed since the day I opened this discussion and start working on the calendar integration for Wino Mail. I am preparing a new version of **Wino Mail 2.0** to be in Store after the beta tests are done, which has integrated Outlook/Gmail API Calendar synchronization and special CalDAV and local calendar support. Key things I want to mention:

- All the work were done in feature/vNext branch for those who are interested.
- Underlying tech and AI era got me to choose different path for doing the development explained in this discussion.
- App is still safe, fast and more stable than ever thanks to those changes.

I have already shared some plans in Discord channel and in [this issue](https://github.com/bkaankose/Wino-Mail/issues/742). From start to finish, almost all parts of the application has been refactored for better.

As I come closer and closer to the finish line with releasing this beast, I want to first close this discussion and cleanup the GitHub issues a little bit according to the results of the beta tests that will go on on the upcoming days. Full changelog with major changes will be released in the repository later on.

Thanks for everybody who joined the discussion, provided feedback and shaped the Wino Mail's future!

**Reply by AkazaRenn** [reply]
Thanks for the update! Does that mean that calendar will be a part of the mail app instead of a standalone one? Any chance we can get the calendar in its own window?

**Reply by bkaankose** [reply]
In separate window is not possible, but you will get 2 app entries as "Wino Mail" and "Wino Calendar" in the Windows taskbar and startup menu.

Both will switch to correct mode in the application itself. Wino Mail is designed to work as single instance app and separation can't be done at this point unfortunately.

**Comment by bkaankose** [comment]
[2.0 beta is released in Github releases. ](https://github.com/bkaankose/Wino-Mail/releases/tag/v2.0) You can join the discussion [here ](https://github.com/bkaankose/Wino-Mail/discussions/843)to give it a try and provide feedback.
